parser.hpp

```diff
@@ -13,6 +13,26 @@
 #include "game_engine.hpp"
 #include "utils.hpp"
 
+struct BoardParameters
+{
+    int row_count;
+    int column_count;
+    int score;
+    int moves;
+};
+
+constexpr int extract_int(auto it)
```

Comment:
As a security it would be nice to take begin and end here (even though this would result in a compile-time crash if we go too far in the buffer :D).

Reply:
Added limit for number of read chars